pkg/steps/multi_stage.go

```diff
@@ -551,11 +555,32 @@ func (s *multiStageTestStep) runPods(ctx context.Context, pods []coreapi.Pod, sh
 }
 
 func (s *multiStageTestStep) runPod(ctx context.Context, pod *coreapi.Pod, notifier *TestCaseNotifier) error {
+	// work around activeDeadlineSeconds vs. container status: https://github.com/kubernetes/kubernetes/issues/88611
+	activeDeadlineSeconds := pod.Spec.ActiveDeadlineSeconds
+	pod.Spec.ActiveDeadlineSeconds = nil
+	// ensure we notice before the pod is deleted
+	pod.ObjectMeta.Finalizers = []string{finalizer}
 	if _, err := createOrRestartPod(s.podClient.Pods(s.jobSpec.Namespace()), pod); err != nil {
 		return fmt.Errorf("failed to create or restart %q pod: %w", pod.Name, err)
 	}
+	podComplete := make(chan struct{}, 1)
+	if activeDeadlineSeconds != nil {
+		go func() {
+			duration := time.Duration(*activeDeadlineSeconds * 1000000000) // convert seconds to nanoseconds
```

Comment:
Not sure this makes sense, the `pod.spec` field should already be in seconds? Why not `* time.Second`?

Reply:
> Why not `* time.Second`?

From the commit message:

> The seconds -> nanoseconds thing bothers me, but:
>
>     time.After(*activeDeadlineSeconds * time.Second)
>
> failed to compile with:
>
>     invalid operation: *activeDeadlineSeconds * time.Second (mismatched types int64 and time.Duration)

Reply:
Obsolete with [the pivot to Vadim's approach][1].

[1]: https://github.com/openshift/ci-tools/pull/1257#issuecomment-705131035